packages/core/src/Connection.test.ts

```diff
@@ -59,22 +58,15 @@ describe('Connection', () => {
 
   it('should send messages to the peer when local state changes', async () => {
     const peer = new WebSocket(url)
-    const _connection = new Connection(repo, peer, fakeDispatch)
-
+    const _ = new Connection(repo, peer, fakeDispatch)
     await _yield()
 
-    expect(peer.send).toHaveBeenCalledWith(JSON.stringify({ type: 'HELLO', documentCount: 1 }))
-
-    expect(peer.send).not.toHaveBeenCalledWith(
-      expect.stringContaining(JSON.stringify({ [localActorId]: 2 }))
-    )
+    expect(peer.send).toHaveBeenCalled()
 
     await repo.change('state', s => (s.boo = 2))
     await _yield()
 
-    expect(peer.send).toHaveBeenCalledWith(
-      expect.stringContaining(JSON.stringify({ [localActorId]: 2 }))
-    )
+    expect(peer.send).toHaveBeenCalled()
   })
 
   it('should call close on peer when close is called', () => {
```

Comment:
These tests were too tightly bound to the `RepoSync`'s implementation details regarding the sequence of messages etc. This class is really just glue between the networking stack and RepoSync, so there's not much to test anyway.